geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -92,6 +95,23 @@ public int llen() {
     return elementList.size();
   }
 
+  /**
+   * @param region the region to set on
+   * @param key the key to set on
+   * @param index the index specified by the user
+   * @param value the value to set
+   */
+  public void lset(Region<RedisKey, RedisData> region, RedisKey key, int index, byte[] value) {
+    int listSize = elementList.size();
+    int adjustedIndex = index >= 0 ? index : listSize + index;
+    if (adjustedIndex > listSize - 1 || adjustedIndex < 0) {
+      throw new RedisException(ERROR_INDEX_OUT_OF_RANGE);
+    }
+
+    elementList.set(adjustedIndex, value);
```

Comment:
you should use elementReplace here instead of elementList.set to give you the synchronization protection from a concurrent toData called by GII